utils/runtime-fuzzer/src/gear_calls.rs

```diff
@@ -0,0 +1,359 @@
+    fn generate(
+        &self,
+        intermediate_data: &mut IntermediateData,
+        unstructured: &mut Unstructured,
+    ) -> Result<GearCall> {
+        let program_id = unstructured
+            .choose(&intermediate_data.uploaded_programs)
+            .copied()?;
+        let payload = arbitrary_payload(unstructured)?;
+        log::trace!(
+            "Random data after payload (send_message) gen {}",
+            unstructured.len()
+        );
+        log::trace!("Payload (send_message) length {:?}", payload.len());
+
+        Ok(SendMessageArgs((program_id, payload, self.gas, self.value, self.prepaid)).into())
+    }
+
+    fn unstructured_size_hint(&self) -> usize {
+        // 512 KiB for payload.
+        520 * 1024
+    }
+}
+
+/// Extrinsic generator that's capable of generating `SendReply` calls.
+pub(crate) struct SendReplyGenerator {
+    pub mailbox_provider: Box<dyn MailboxProvider>,
+
+    pub gas: u64,
+    pub value: u128,
+    pub prepaid: bool,
+}
+
+impl ExtrinsicGenerator for SendReplyGenerator {
+    fn generate(
+        &self,
+        _: &mut IntermediateData,
+        unstructured: &mut Unstructured,
+    ) -> Result<GearCall> {
+        log::trace!(
+            "Random data before payload (send_reply) gen {}",
+            unstructured.len()
+        );
+        let message_id =
+            arbitrary_message_id_from_mailbox(unstructured, self.mailbox_provider.as_ref())?;
+
+        let payload = arbitrary_payload(unstructured)?;
+        log::trace!(
+            "Random data after payload (send_reply) gen {}",
+            unstructured.len()
+        );
+        log::trace!("Payload (send_reply) length {:?}", payload.len());
+
+        Ok(SendReplyArgs((message_id, payload, self.gas, self.value, self.prepaid)).into())
+    }
+
+    fn unstructured_size_hint(&self) -> usize {
+        // 512 KiB for payload.
+        520 * 1024
+    }
+}
+
+/// Extrinsic generator that's capable of generating `ClaimValue` calls.
+pub(crate) struct ClaimValueGenerator {
+    pub mailbox_provider: Box<dyn MailboxProvider>,
+}
+
+impl ExtrinsicGenerator for ClaimValueGenerator {
+    fn generate(
+        &self,
+        _: &mut IntermediateData,
+        unstructured: &mut Unstructured,
+    ) -> Result<GearCall> {
+        log::trace!("Generating claim_value call");
+        let message_id =
+            arbitrary_message_id_from_mailbox(unstructured, self.mailbox_provider.as_ref())?;
+
+        Ok(ClaimValueArgs(message_id).into())
+    }
+
+    fn unstructured_size_hint(&self) -> usize {
+        // 32 bytes for message id.
+        100
+    }
+}
+
+fn arbitrary_message_id_from_mailbox(
+    u: &mut Unstructured,
+    mailbox_provider: &dyn MailboxProvider,
+) -> Result<MessageId> {
+    let messages = mailbox_provider.fetch_messages();
+
+    if messages.is_empty() {
+        log::trace!("Mailbox is empty. Selecting random message id");
+        arbitrary_message_id(u)
```

Comment:
I suggest not even generating `SendReply` when there are no messages in the mailbox